core/trino-main/src/main/java/io/trino/operator/aggregation/ApproximateCountDistinctAggregation.java

```diff
@@ -123,14 +128,22 @@ public static void input(
     {
         HyperLogLog hll = getOrCreateHyperLogLog(state, maxStandardError);
         state.addMemoryUsage(-hll.estimatedInMemorySize());
-        long hash;
         try {
-            hash = (long) methodHandle.invoke(value);
+            if (type instanceof StreamType streamType) {
```

Comment:
The semantics of `approx_distinct` aggregation is that it evaluates # distinct input values of some type.
If a user wrote `approx_distinct(stream(a))`, it should be # of distinct streams. 

Here, we are changing the aggregate function behavior completely based on the input type. Instead, to avoid ambiguity of this kind, we should actually have another aggregate function, say `stream_approx_distinct(a)` that takes an array and processes it element-wise. Such approach would also save us from introducing an artificial type and a `stream()` function. 

We could further think of a syntactic sugar like `sum(a STREAMED)` which would be resolved to `stream_sum(a)`.

Reply:
Note that adding new built-in aggregate functions needs to be approved language-wise same as adding the `STREAM` syntax.

Reply:
@kasiafi, 
I like `APPROX_DISTINCT(a STREAMED)` because it enables exactly same simplicity and elegance.

Is it considered adding a new keyword to SQL? OR is it considered a parameter into the function?
If later than I can see the strength of the proposal.

Context for others: Continuing the discussion from https://github.com/trinodb/trino/issues/22445#issuecomment-2515316091

Reply:
@cploonker `STREAMED` would be a keyword. Similar to `DISTINCT` in `sum(DISTINCT a)`.

After giving it a second glance, I'm not convinced it is a good idea to add it. This modifier would change the `approx_distinct` functions return type and its behavior, while such modifiers are only meant to pre-process the input in some way.

That said, what about adding separate aggregate functions, like `stream_approx_distinct(a)`? Even if they don't make their way to builtin, you can keep them in a plugin.

Reply:
@kasiafi 
I agree with you on not adding a new keyword.

Let us try to brainstorm a little more if we can come up with a creative solution.

Given `UNNEST` is already a keyword, how bad is it to create a new context dependent usage for it. 
e.g  `approx_distinct(UNNEST a)`
Here the result of UNNEST is not joined to the original table and just fed into the function.
Please pardon me if I am committing blasphemy here. If so, please forget I said this. I will delete it from my comment if necessary.


@felix-alonso, just in case if you have any other ideas for syntax.